### DISCUSSION TITLE
Support for disabling PSRAM

### DISCUSSION THREAD

**Discussion by jeroenst** [benzino77/tasmocompiler#428] [Ideas]
Is it possible to add a build flag so the compiler builds without PSRAM enabled causing relays to click on boot?

https://github.com/arendst/Tasmota/pull/21282#issuecomment-2549312242

**Comment by benzino77** [comment]
No, it is not possible to without some TasmoCompiler feature/board definition tinkering, but as you can see [here](https://github.com/benzino77/tasmocompiler/blob/01b64d05dd3f1d9e27ec11a2112a8d9fd5fb1ade/src/components/AppStepper/FeaturesStep/AvailableFeatures.js#L443), some features adds entries to platformio file. 

You can try to define your own feature
```js
{
    name: 'jeroenst_feature',
    value: true,
    show: true,
    description: 'stepFeaturesBluetoothDesc',
    tooltip: 'stepFeaturesBluetoothTooltip',
    platformio_entries: {
      build_unflags: ['${env:tasmota32_base.build_unflags}', '-DBOARD_HAS_PSRAM'],
      build_flags: ['${env:tasmota32.build_flags}', '-DDISABLE_PSRAMCHECK'],
    },
    custom: '',
    boards: ['all'],
},
```

But I haven't tested it. I would rather suggest to compile your firmware using native platformio environment.

**Comment by jeroenst** [comment]
Can an option NOPSRAM be added like this so it's selectable for all users?

**Reply by benzino77** [reply]
If there will be lot of users interested in such a feature...

**Reply by jeroenst** [reply]
If they want to use the dingtian 4 channel relay board they need this feature.

However when it's to much hassle to add it I will compile it once myself with platformio.
